sct-commons/src/main/java/org/lfenergy/compas/sct/commons/util/Utils.java

```diff
@@ -386,4 +395,62 @@ public static boolean isExtRefFeedBySameControlBlock(TExtRef t1, TExtRef t2) {
                 && Utils.equalsOrBothBlank(t1.getSrcCBName(), t2.getSrcCBName())
                 && Objects.equals(t1.getServiceType(), t2.getServiceType());
     }
+
+    /**
+     * provides Active LDevice for LDEPF setting's inst attribute
+     */
+    public static final BiFunction<IEDAdapter, LDEPFSettingData, Optional<LDeviceAdapter>> getActiveSourceLDevice = (iedAdapter, setting) ->
+            iedAdapter.findLDeviceAdapterByLdInst(setting.getLdInst())
+                    .filter(lDeviceAdapter -> lDeviceAdapter.getLDeviceStatus()
+                            .map(status -> status.equals(LDeviceStatus.ON))
+                            .orElse(false))
+                    .stream().findFirst();
+
+    /**
+     * provides Active LN Object that satisfies the LDEPF settings attributes (lnClass, lnInst, prefix)
+     */
+    public static final BiFunction<LDeviceAdapter, LDEPFSettingData, Optional<AbstractLNAdapter<?>>> getActiveLNodeSource = (lDeviceAdapter, setting) ->
+            lDeviceAdapter.getLNAdaptersIncludingLN0().stream()
+                    .filter(lnAdapter -> lnAdapter.getLNClass().equals(setting.getLnClass())
+                            && lnAdapter.getLNInst().equals(setting.getLnInst())
+                            && Utils.equalsOrBothBlank(setting.getLnPrefix(), lnAdapter.getPrefix()))
+                    .findFirst()
+                    .filter(lnAdapter -> lnAdapter.getDaiModStValValue()
+                            .map(status -> status.equals(LDeviceStatus.ON))
+                            .orElse(true));
+
+
+    /**
+     * verify whether the LN satisfies the LDEPF settings for Data Type Template elements.
+     */
+    public static final BiPredicate<AbstractLNAdapter<?>, LDEPFSettingData> isValidDataTypeTemplate = (lnAdapter, setting) -> {
```

Comment:
It's better to use a function :
```java
public static boolean isValidDataTypeTemplate(LDEPFSettingData setting, AbstractLNAdapter<?> lnAdapter) {
```
But we can do that later if you want

Reply:
done